refined procedures (#644)
openstack-k8s-operators/data-plane-adoption: docs_user/assemblies/assembly_configuring-isolated-networks.adoc

= Configuring isolated networks

> Before you begin replicating your existing VLAN and IPAM configuration in the {rhos_long} environment, you must have the following IP address allocations for the new control plane services:

Author: @booxter Does the sentence make sense as to what the customer is going to configure in this assembly?

"Before you begin replicating your existing VLAN and IPAM configuration in the {rhos_long} environment, you must have the following IP address allocations for the new control plane services:"

* 1 IP address, per isolated network, per {rhocp_long} worker node. These IP addresses configure {OpenShiftShort} worker nodes to the `NodeNetworkConfigurationPolicy` custom resources (CRs). For more information, see xref:configuring-openshift-worker-nodes_isolated-networks[Configuring {OpenShiftShort} worker nodes].
* 1 IP range, per isolated network, for the data plane nodes. These ranges
configure data plane nodes to `NetConfig` CRs. For more information, see xref:configuring-data-plane-nodes_isolated-networks[Configuring data plane nodes].
* 1 IP range, per isolated network, for control plane services. These ranges
enable pod connectivity to isolated networks to
`NetworkAttachmentDefinition` CRs. For more information, see xref:configuring-networking-for-control-plane-services_isolated-networks[Configuring the networking for control plane services].

> * 1 IP range, per isolated network, for load balancer IP addresses. These IP ranges define load balancer IP addresses for MetalLB in the `IPAddressPool` CRs. For more information, see xref:configuring-networking-for-control-plane-services_isolated-networks[Configuring the networking for control plane services].

Author: I have a couple of questions on the bulleted list:

1. Is it correct that they need 1 IP range per isolated network for data plane nodes, control plane services, and load balancer IP addresses?
2.  Do the IP addresses really "configure" the worker nodes to the custom resource? Or do you mean to say “You configure these IP addresses in the `NodeNetworkConfigurationPolicy` custom resources (CRs) for the OpenShift worker nodes”? Similar question for the following bullet points, e.g., should the second bullet point say "You configure IP ranges in the `NetConfig` CRs for the data plane nodes"?

[NOTE]
The exact list and configuration of isolated networks in this procedure should reflect the actual adopted environment. The number of isolated networks might differ from the examples used in this procedure. IPAM scheme might also differ.
Only relevant parts of the configuration are shown. Examples are incomplete and
should be incorporated into the general configuration for the new deployment,

> as described in the general {rhos_prev_long} documentation.

Author: A couple of questions on this note:

1. I’m a little bit confused by the sentence “Only relevant parts of the configuration are shown.” Does this mean the relevant parts to the adoption procedure?
2. In the last sentence in the note,  are we trying to say that the examples don’t necessarily reflect the customers’ configuration, and they need to use values that are specific to their configuration?
3. Does “adopted environment” refer to the RHOSO environment?

= Configuring isolated networks

Before you begin replicating your existing VLAN and IPAM configuration in the {rhos_long} environment, you must have the following IP address allocations for the new control plane services:

> * 1 IP address, per isolated network, per {rhocp_long} worker node. You configure these IP addresses in the `NodeNetworkConfigurationPolicy` custom resources (CRs) for the {OpenShiftShort} worker nodes. For more information, see xref:configuring-openshift-worker-nodes_isolated-networks[Configuring {OpenShiftShort} worker nodes].

Reviewer: Do you mean:

```suggestion
* 1 IP address for each isolated network on each {rhocp_long} worker node. You configure these IP addresses in the `NodeNetworkConfigurationPolicy` custom resources (CRs) for the {OpenShiftShort} worker nodes. For more information, see xref:configuring-openshift-worker-nodes_isolated-networks[Configuring {OpenShiftShort} worker nodes].
```

> * 1 IP range, per isolated network, for the data plane nodes. You configure these ranges in the `NetConfig` CRs for the data plane nodes. For more information, see xref:configuring-data-plane-nodes_isolated-networks[Configuring data plane nodes].

Reviewer: ```suggestion
* 1 IP range for each isolated network for the data plane nodes. You configure these ranges in the `NetConfig` CRs for the data plane nodes. For more information, see xref:configuring-data-plane-nodes_isolated-networks[Configuring data plane nodes].
```

> * 1 IP range, per isolated network, for control plane services. These ranges

Reviewer: ```suggestion
* 1 IP range for each isolated network for control plane services. These ranges
```

enable pod connectivity for isolated networks in the `NetworkAttachmentDefinition` CRs. For more information, see xref:configuring-networking-for-control-plane-services_isolated-networks[Configuring the networking for control plane services].

> * 1 IP range, per isolated network, for load balancer IP addresses. These IP ranges define load balancer IP addresses for MetalLB in the `IPAddressPool` CRs. For more information, see xref:configuring-networking-for-control-plane-services_isolated-networks[Configuring the networking for control plane services].

Reviewer: ```suggestion
* 1 IP range for each isolated network for load balancer IP addresses. These IP ranges define load balancer IP addresses for MetalLB in the `IPAddressPool` CRs. For more information, see xref:configuring-networking-for-control-plane-services_isolated-networks[Configuring the networking for control plane services].
```
